scipy/stats/_distn_infrastructure.py

```diff
@@ -3392,7 +3392,7 @@ def _cdf_single(self, k, *args):
         return np.sum(self._pmf(m, *args), axis=0)
 
     def _cdf(self, x, *args):
-        k = floor(x)
+        k = floor(x).astype(np.float64)
```

Comment:
I've tested this change locally. The previously failing SymPy test passes with this change so it looks good to me.